File: go/1725.go
func countGoodRectangles(rectangles [][]int) int {
    ans := 0
    cnt := 0
    for _, rect := range rectangles {
        l := 0
        if rect[0] < rect[1] {
            l = rect[0]
        } else {
            l = rect[1]
        }
        switch {
            case l == ans:
                cnt += 1
            case l > ans:
                ans = l
                cnt = 1
            default:
                //
        }
    }
    return cnt
}

File: rust/1725.rs
impl Solution {
    pub fn count_good_rectangles(rectangles: Vec<Vec<i32>>) -> i32 {
        let mut ans = 0;
        let mut cnt = 0;
        for rect in rectangles.iter() {
            let l = std::cmp::min(rect[0], rect[1]);
            if l == ans {
                cnt += 1;
            } else if l > ans {
                ans = l;
                cnt = 1;
            }
        }
        cnt
    }
}
